src/main/java/pl/futurecollars/invoicing/service/TaxCalculatorService.java

```diff
@@ -0,0 +1,65 @@
+package pl.futurecollars.invoicing.service;
+
+import java.math.BigDecimal;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import pl.futurecollars.invoicing.db.Database;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.model.InvoiceEntry;
+import pl.futurecollars.invoicing.model.TaxReport;
+
+@RequiredArgsConstructor
+@Service
+public class TaxCalculatorService {
+
+    private final Database database;
+
+    private BigDecimal visit(Predicate<Invoice> predicate, Function<InvoiceEntry, BigDecimal> function) {
+        return database.getAll()
+            .stream()
+            .filter(predicate)
+            .flatMap(invoice -> invoice.getEntries().stream())
+            .map(function)
+            .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    public BigDecimal income(String taxIdentificationNumber) {
```

Comment:
I would change all access modifiers to private and leave only taxReport() method public